dask/array/core.py

```diff
@@ -929,13 +936,21 @@ def store(sources, targets, lock=True, regions=None, compute=True,
         store_dsks, tgt_dsks, [sources_dsk]
     ))
 
+    from ..optimization import cull
+    store_dsks_mrg = cull(store_dsks_mrg, store_keys)[0]
+
     if return_stored:
+        load_dsks_mrg = sharedict.merge(*load_dsks)
+
         if compute:
             store_dlyds = [Delayed(k, store_dsks_mrg) for k in store_keys]
             store_dlyds = persist(*store_dlyds)
             store_dsks_mrg = sharedict.merge(*[e.dask for e in store_dlyds])
 
-        load_dsks_mrg = sharedict.merge(store_dsks_mrg, *load_dsks)
+        load_dsks_mrg = sharedict.merge(load_dsks_mrg, store_dsks_mrg)
+
+        from ..optimization import inline
```

Comment:
Trying to import this outside the function results in an `ImportError` during testing. So imported it locally only where it is needed to avoid this issue.